pandas/tests/test_index.py

```diff
@@ -2812,6 +2812,20 @@ def test_view(self):
         result = self._holder(i)
         tm.assert_index_equal(result, i)
 
+class DatetimeAbsoluteLike(DatetimeLike):
+
+    def test_contains(self):
+
+        i = self.create_index()
+
+        self.assertTrue(i[2] in i)
+        self.assertFalse('2012' in i)
+        #  python datetime objects work
+        self.assertTrue(datetime(2013,1,1) in i)
+        # strings work
+        self.assertTrue('2013-1-1' in i)
+
+
 class TestDatetimeIndex(DatetimeLike, tm.TestCase):
```

Comment:
I don't think this actually running
u need to add this class as a mixin to the test classes for DatetimeIndex/Periodindex